doc/developers/contributing.rst

```diff
@@ -188,6 +188,54 @@ Contributing code
 How to contribute
 -----------------
 
+Before you start contributing, you need to setup your development environment.
+This can be either a virtual environment with Conda or your own primary 
+development environment.
+
+Using a Virtual Environment with Conda
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Using a virtual environment allows you to seperate your working directory and
+its packages from your primary development environment. This way, the unstable 
+scikit-learn library in development mode does not temper with your system's 
+scikit-learn's performance.
+
+1. `Download the Anaconda Distribution <https://www.anaconda.com/distribution/>`__ for your OS. 
+   Make sure you download Python 3.6+ version since scikit-learn only supports 
+   the newer versions of Python 3.
+2. Open the Anaconda Command Prompt
+3. Create a virtual environment for scikit-learn development::
+
+       $ conda create -n sklearndev numpy scipy matplotlib pytest sphinx cython ipykernel sphinx-gallery
+
+   Here, you can replace the sklearndev with any name you like. 
+   If conda doesn't find any library mentioned above, remove it from the command 
+   and proceed to the next steps and install it using pip.
+
+4. Activate the virtual environment::
+
+       $ conda activate sklearndev
+
+5. Install any additional packages using pip::
+
+       $ pip install package-name
+
+Using your primary development environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Just install the packages needed for testing::
+
+       $ pip install cython pytest flake8
```

Comment:
I can't think of a case where it makes sense to use the system python env for development. Maybe also suggesting using virtualenv? It is built in in CPython 3.3+.

````
python -m venv sklearn-env
```
then
```
source sklearn-env/bin/activate  # on Unix
.\sklearn-env\Scripts\activate.bat # on Windows

continue with pip install as usual
```

Reply:
Also the above makes is sounds as though vitualenv are somehow conda specific -- it is not (though granted it has more capability there).